packages/ui/src/bounty/components/BountyPreviewHeader/BountyPreviewHeader.tsx

```diff
@@ -175,17 +196,11 @@ const SuccessfulStageButtons = React.memo(({ bounty, activeMember, t }: BountyHe
   )
 })
 
-const FailedStageButtons = React.memo(({ bounty, activeMember, t }: BountyHeaderButtonsProps) => {
-  const isContributor = useMemo(
-    () => bounty.contributors?.some((contributor) => contributor.actor?.id === activeMember?.id),
-    [bounty]
-  )
-
-  const userEntry = useMemo(() => bounty.entries?.find((entry) => entry.worker.id === activeMember?.id), [bounty])
+const FailedStageButtons = React.memo(({ bounty, t, userEntry, isContributor }: BountyHeaderButtonsProps) => {
   const hasAnnounced = !!userEntry
   const hasSubmitted = hasAnnounced && userEntry.hasSubmitted
   const hasLost = hasSubmitted && !userEntry.winner && !userEntry.rejected && !bounty.isTerminated
-  const isTerminated = !bounty.isTerminated
+  const isTerminated = !bounty.isTerminated && isContributor
 
   if (!hasAnnounced && !isContributor) {
     return null
```

Comment:
This: `isTerminated = !bounty.isTerminated ...` is kind of confusing :raised_eyebrow: 
What about ?
```suggestion
  if (bounty.isTerminated || !hasAnnounced && !isContributor) {
    return null
```